sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/physical/partitioning.scala

```diff
@@ -72,9 +72,14 @@ case object AllTuples extends Distribution {
 /**
  * Represents data where tuples that share the same values for the `clustering`
  * [[Expression Expressions]] will be co-located in the same partition.
+ *
+ * @param requireAllClusterKeys When true, `Partitioning` which satisfies this distribution,
+ *                              must match all `clustering` expressions in the same ordering.
  */
 case class ClusteredDistribution(
     clustering: Seq[Expression],
+    requireAllClusterKeys: Boolean = SQLConf.get.getConf(
```

Comment:
nit: it's less breaking to put new parameter at the end, so that some code can remain unchanged.

Reply:
@cloud-fan - I agree for the point of caller-side code unchanged. I guess it's just feeling more coherent for others to read and understand code, when putting `clustering` and `requireAllClusterKeys` together. This was raised by https://github.com/apache/spark/pull/35574#discussion_r813499279 by @HeartSaVioR  as well. I am curious would adding the field in the middle here break other external library depending on Spark? I guess otherwise reviewers already paid the cost to review this PR, so not sure how important to change the caller-side code back. Just want to understand more here.

Reply:
OK maybe it's not a big deal